[2025-04-24T18:57:33.020+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-04-24T18:57:33.177+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: perbaikan_skripsi.wait_for_new_data scheduled__2025-04-24T18:55:00+00:00 [queued]>
[2025-04-24T18:57:33.323+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: perbaikan_skripsi.wait_for_new_data scheduled__2025-04-24T18:55:00+00:00 [queued]>
[2025-04-24T18:57:33.332+0000] {taskinstance.py:2866} INFO - Starting attempt 2 of 4
[2025-04-24T18:57:33.701+0000] {taskinstance.py:2889} INFO - Executing <Task(SqlSensor): wait_for_new_data> on 2025-04-24 18:55:00+00:00
[2025-04-24T18:57:33.759+0000] {logging_mixin.py:190} WARNING - /home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=69894) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-04-24T18:57:33.763+0000] {standard_task_runner.py:72} INFO - Started process 70009 to run task
[2025-04-24T18:57:33.807+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'perbaikan_skripsi', 'wait_for_new_data', 'scheduled__2025-04-24T18:55:00+00:00', '--job-id', '22879', '--raw', '--subdir', 'DAGS_FOLDER/tugas_akhir.py', '--cfg-path', '/tmp/tmpmj9zs4s5']
[2025-04-24T18:57:33.832+0000] {standard_task_runner.py:105} INFO - Job 22879: Subtask wait_for_new_data
[2025-04-24T18:57:34.346+0000] {task_command.py:467} INFO - Running <TaskInstance: perbaikan_skripsi.wait_for_new_data scheduled__2025-04-24T18:55:00+00:00 [running]> on host a38b69ba8f8a
[2025-04-24T18:57:36.596+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Dion Rikki' AIRFLOW_CTX_DAG_ID='perbaikan_skripsi' AIRFLOW_CTX_TASK_ID='wait_for_new_data' AIRFLOW_CTX_EXECUTION_DATE='2025-04-24T18:55:00+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-04-24T18:55:00+00:00'
[2025-04-24T18:57:36.613+0000] {logging_mixin.py:190} INFO - Task instance is in running state
[2025-04-24T18:57:36.617+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2025-04-24T18:57:36.624+0000] {logging_mixin.py:190} INFO - Current task name:wait_for_new_data state:running start_date:2025-04-24 18:57:33.181065+00:00
[2025-04-24T18:57:36.627+0000] {logging_mixin.py:190} INFO - Dag name:perbaikan_skripsi and current dag run status:running
[2025-04-24T18:57:36.631+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-04-24T18:57:36.842+0000] {base.py:84} INFO - Retrieving connection 'mysql_default'
[2025-04-24T18:57:36.941+0000] {sql.py:101} INFO - Poking: 
            SELECT COUNT(*) 
            FROM tugas_akhir_sensordata s
            WHERE s.timestamp >= NOW() - INTERVAL 3 MINUTE
            AND s.soil_type IS NOT NULL
            AND s.crop_type IS NOT NULL
            AND NOT EXISTS (
                SELECT 1 
                FROM fertilizer_predictions f
                WHERE f.sensor_data_id = s.id
            )
         (with parameters None)
[2025-04-24T18:57:36.983+0000] {base.py:84} INFO - Retrieving connection 'mysql_default'
[2025-04-24T18:57:37.042+0000] {sql.py:553} INFO - Running statement: 
            SELECT COUNT(*) 
            FROM tugas_akhir_sensordata s
            WHERE s.timestamp >= NOW() - INTERVAL 3 MINUTE
            AND s.soil_type IS NOT NULL
            AND s.crop_type IS NOT NULL
            AND NOT EXISTS (
                SELECT 1 
                FROM fertilizer_predictions f
                WHERE f.sensor_data_id = s.id
            )
        , parameters: None
[2025-04-24T18:57:37.057+0000] {sql.py:562} INFO - Rows affected: 1
[2025-04-24T18:57:48.347+0000] {taskinstance.py:309} INFO - Rescheduling task, marking task as UP_FOR_RESCHEDULE
[2025-04-24T18:57:48.348+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-04-24T18:57:48.357+0000] {job.py:229} INFO - Heartbeat recovered after 16.02 seconds
[2025-04-24T18:57:48.603+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-04-24T18:57:48.908+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-04-24T18:57:48.915+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
[2025-04-24T18:57:49.033+0000] {standard_task_runner.py:217} INFO - Process not found (most likely exited), stop collecting metrics
